packages/expo-barcode-scanner/src/BarCodeScanner.tsx

```diff
@@ -0,0 +1,151 @@
+import { UnavailabilityError } from 'expo-errors';
+import mapValues from 'lodash.mapvalues';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { findNodeHandle, Platform, ViewProps, ViewPropTypes } from 'react-native';
+
+import ExpoBarCodeScannerModule from './ExpoBarCodeScannerModule';
+import ExpoBarCodeScannerView from './ExpoBarCodeScannerView';
+
+const { BarCodeType, Type } = ExpoBarCodeScannerModule;
+
+const EVENT_THROTTLE_MS = 500;
+
+type BarCodeEvent = {
+  type: string;
+  data: string;
+  [key: string]: any;
+};
+
+export type BarCodeEventCallbackArguments = {
+  nativeEvent: BarCodeEvent;
+};
+
+export type BarCodeReadCallback = (params: BarCodeEvent) => void;
+
+export type BarCodeScannedCallback = (params: BarCodeEventCallbackArguments) => void;
```

Comment:
We pass `objectFromTheNativeLayer.nativeEvent` into either the "read" or "scanned" callback -> I don't think `objectFromTheNativeLayer` has `nativeEvent.nativeEvent`:

```suggestion
export type BarCodeScannedCallback = (params: BarCodeEvent) => void;
```